### DISCUSSION TITLE
Add 2FA or passkeys for user login

### DISCUSSION THREAD

**Discussion by jasperrooduijn** [ananthakumaran/paisa#131] [Ideas]
Thank you for your dedication and hard work in developing Paisa. Your efforts are greatly appreciated. The app really helps me visualising data from my ledger-cli instance to my partner. We are using the app with our family and not all users are very tech savvy. Thats why i would like to host the app on one of my servers, so all multiple members of the family can use it (read only mode). Adjustments to the journal will be done by a script and versioning will be used to keep track. We use the user account in your app for security and It is also protected by HTTP Basic Authentication.

But still i would like to suggest the integration of Two-Factor Authentication (2FA) or Passkeys into your app. This addition is crucial for enhancing user account security when serving Paisa from a public server. Implementing 2FA or Passkeys would provide an essential layer of protection against unauthorized access.

Let me know your toughts.

**Comment by ananthakumaran** [comment]
2FA/Passkeys/other types of login mechanisms are possible. I will likely not add any external dependency (that rules out thinks like SMS 2FA etc), but anything that can be implemented in a self-contained manner (TOTP like) can be done. That said, this is not a high priority and I might take some time to research/implement it.

For now, I would suggest that you keep it out of the public internet as much as possible. There are services like [tailscale](https://tailscale.com/pricing) which allow you to share a domain with others and support SSO login. If that's not possible, use a strong password.

**Reply by jasperrooduijn** [reply]
Thanks! It would be very nice if some kind of 2FA auth would make it in to your app, a TOTP method would also be fine for me. I fully understand this has no high prio. I'll dive into the tailscale tip, thanks for that.

**Reply by avinal** [reply]
I am using tailscale to access most services on my home server. IMO even if the application provides me a secure way to expose to public internet, I will likely keep it local only unless really required. Tailscale works best.

**Reply by jasperrooduijn** [reply]
Nice! Thanks for the conformation. I will setup something like this soon.
